src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,572 @@
+        start_date: date = first_available_date,
+        end_date: date = None,
+    ) -> xr.Dataset:
+        """
+        Load the CHIRPS data.
+
+        Should only be called after the data
+        has been downloaded and processed.
+
+        Parameters
+        ----------
+        start_date: int, default = None
+            Data will be loaded starting from date `start_date`.
+            If None, it is set to 1981-1-1.
+        end_date: int, default = None
+            Data will be loaded up to date `end_date`.
+            If None, it is set to the date for which most recent data
+            is available.
+
+        Returns
+        -------
+        The processed CHIRPS dataset.
+        """
+        # Get list of filepaths of files to be loaded
+        if end_date is None:
+            end_date = self._get_last_available_date()
+
+        filepath_list = self._get_to_be_loaded_path_list(
+            start_date=start_date,
+            end_date=end_date,
+        )
+
+        # Merge all files in one dataset
+        if not filepath_list:
+            raise FileNotFoundError(
+                "Cannot find any netcdf file for the chosen combination "
+                "of frequency, resolution and area. Make sure "
+                "sure that you have already called the 'process' method."
+            )
+
+        try:
+            ds = xr.open_mfdataset(
+                filepath_list,
+            )
+            # include the names of all files that are included in the ds
+            ds.attrs["included_files"] = [f.stem for f in filepath_list]
+        except FileNotFoundError as err:
+            raise FileNotFoundError(
+                "Cannot find one or more netcdf files corresponding "
+                "to the selected range. Make sure that you already "
+                "downloaded and processed those data."
+            ) from err
+
+        return ds.rio.write_crs("EPSG:4326", inplace=True)
+
+    def _download_prep(self, d, clobber):
+        # Preparatory steps for the actual download
+        year = str(d.year)
+        month = f"{d.month:02d}"
+        day = f"{d.day:02d}"
+        output_filepath = self._get_raw_path(year=year, month=month, day=day)
+        output_filepath.parent.mkdir(parents=True, exist_ok=True)
+        url = self._get_url(year=year, month=month, day=day)
+        # Actual download
+        return self._download(
+            filepath=output_filepath,
+            url=url,
+            clobber=clobber,
+        )
+
+    def _create_date_list(self, start_date, end_date):
+        """Create list of tuples containing the range of dates of interest."""
+        self._check_dates_validity(
+            start_date=start_date,
+            end_date=end_date,
+        )
+
+        date_list = pd.date_range(
+            start_date, end_date, freq=frequency_dict[self._frequency]
+        ).tolist()
+
+        # Create a sentence containing information on the downloaded data
+        sentence_to_print = (
```

Comment:
Ah I see. I would still try to avoid the message passing: it's kind of confusing when methods return two different things, also it means that `_get_to_be_loaded_path_list` doesn't actually use the message. 

What you could do instead is either have a boolean flag in `_create_date_list` to print the message or not. Even better, pass the logging level that you want to use to print the message. You could make the default debug, and `download` would pass `info`.

Reply:
Sorry, I don't understand the last part.

Reply:
No worries, it was not super clear. What you could do is make one of the parameters to `_create_date_list` be `logging_level`, and then you can call `logging.log(logging_level, msg)`.